Remove u prefix from universal-ctags binary

gpanders opened the pull request:
#### Description

<!-- Note: it is best to make pull requests from a branch rather than from master -->

Install universal-ctags as `${prefix}/bin/ctags`. Universal Ctags is designed to be a drop-in replacement for Exuberant Ctags. Installing it as `uctags` defeats the point.

This reverts the change from 0b86afc1b91a5f12b8eabd8499c3cb87c3d5d02b which was meant to address a conflict in the `global` port. To address that conflict, provide `+ctags` and `+universal_ctags` variants for `global` to use *either* Exuberant Ctags *or* Universal Ctags, not both.

(Note that I did try to simply use `depends_lib bin:ctags:ctags` in the global port and allow the configure script to pick up which version of Ctags is installed. Unfortunately, the global configure script has a bug: it checks to see if `ctags` is Exuberant Ctags using `ctags --version | grep 'Exuberant Ctags'` which doesn't work because the Universal Ctags `--version` information *also* includes the phrase "Exuberant Ctags". So the configure script mistakenly assumes Exuberant is installed even when it is not.)

See also: https://trac.macports.org/ticket/60563

If this approach is undesirable then I propose implementing `ctags_select` to allow the user to choose which version of ctags `${prefix}/bin/ctags` links to while still allowing multiple versions to be installed.

###### Type(s)
<!-- update (title contains ": U(u)pdate to"), submission (new Portfile) and CVE Identifiers are auto-detected, replace [ ] with [x] to select -->

- [ ] bugfix
- [X] enhancement
- [ ] security fix

###### Tested on
<!-- Triple-click and copy the next line and paste it into your shell. It will copy your OS and Xcode version to the clipboard. Paste it here replacing this section.
printf "%s\n" "macOS $(sw_vers -productVersion) $(sw_vers -buildVersion)" "$(xcodebuild -version|awk 'NR==1{x=$0}END{print x" "$NF}')"|tee /dev/tty|pbcopy
-->
macOS 10.15.7 19H524
Xcode 12.4 12D4e


###### Verification <!-- (delete not applicable items) -->
Have you

- [X] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [X] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [X] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [X] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? <!-- Please don't open a new Trac ticket if you are submitting a pull request. -->
- [X] checked your Portfile with `port lint`?
- [X] tried existing tests with `sudo port test`?
- [X] tried a full install with `sudo port -vst install`?
- [X] tested basic functionality of all binary files?

<!-- Use "skip notification" (surrounded with []) to avoid notifying maintainers -->

gpanders commented:
ping @kaimingguo

szhorvat commented:
Good to note that the `ctags` port installs `libreadtags` (as readtags.h and readtags.o), while this port does not. In order for this to be a true drop-in replacement, it should also install `libreadtags` (which is hopefully binary compatible between the two ctags implementations). Or perhaps libreadtags could be broken out into a separate port.

gpanders commented:
> Or perhaps libreadtags could be broken out into a separate port.

universal-ctags packages libreadtags separately: https://github.com/universal-ctags/libreadtags. So making it a separate port (or at least a subport) is probably a good idea.

szhorvat commented:
Making it a separate port would also fix this licensing issue: #10842.

gpanders commented:
I just rebased this branch back onto master which should re-trigger the CI. I also removed the changes to xemacs since that port appears to be broken (it doesn't build on my machine or on the CI's, either on master or this branch).

reneeotten commented:
I have not really an opinion on this change, but there needs to be a decision what to do with this PR. Either we merge it or close it - please respond here with thoughts/opinions; if nothing happens I'll close this in a few days.

gpanders commented:
@reneeotten Let's just go ahead and close it. I've found a workaround in the meantime and I don't have the bandwidth to deal with any breakages this (might) cause down the road.
